Automattic/wp-calypso
# Remove unused logo mutation hook

### Proposed Changes

Remove the [useSiteLogoMutation](https://github.com/Automattic/wp-calypso/blob/faf7992e0e93c86144370a0551d3a22e88555016/packages/data-stores/src/queries/use-site-logo-mutation.ts) hook, which is unused. It was replaced with the [useAndSetSiteLogo](https://github.com/Automattic/wp-calypso/blob/faf7992e0e93c86144370a0551d3a22e88555016/packages/onboarding/src/upload-and-set-site-logo.ts) hook.

### Testing Instructions

The hook being removed was not in use, so this change should have no effects. Even so, I would recommend two tests: 

1) Checkout this branch and search the codebase for useSiteLogoMutation (the hook from the deleted file) and SetSiteLogoResponse (an exported type/interface from the deleted file) and confirm that neither are used anywhere. 

2) Test a startup flow to confirm no regressions.
* Go to https://wordpress.com/hp-2022-tailored-flows/ and select the Newsletter or Link in Bio flow.
* Before going to the site setup page, replace wordpress.com with calypso.localhost:3000 so you are running your local version of calypso. 
* Walk through the usual setup process. When you arrive at the Launchpad screen, confirm that the site logo is correctly reflected in the web preview. That will mean that site icon/logo behaviors are still as expected.
